packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -7153,7 +7153,83 @@ class KeyedSubtree extends StatelessWidget {
   Widget build(BuildContext context) => child;
 }
 
-/// A platonic widget that calls a closure to obtain its child widget.
+/// A platonic (simple) widget that provides a specific [BuildContext] object to its child
+/// widget.
+///
+/// This [BuildContext] object is unique and is not the same object as
+/// the one provided by the framework in the `Widget build` method.
+///
+/// Why is this important? Take the following example, where we would like to
+/// show a [SnackBar] widget to a user when they tap a button:
+///
+/// {@tool snippet}
+/// Example of Scaffold.of With No Builder
+/// ```dart
+///  // BAD EXAMPLE - DO NOT USE
+///  Widget build(BuildContext context) {
+///    return Scaffold(
+///      appBar: AppBar(
+///        title: Text('Scaffold.of Example'),
+///      ),
+///      body: Center(
+///        child: RaisedButton(
+///          child: Text('Show a SnackBar'),
+///          onPressed: () {
+///            // This will fail because the context is from our build method
+///            // and not the RaisedButton
+///            Scaffold.of(context).showSnackBar(
+///              SnackBar(
+///                content: Text('Have a snack!'),
+///              ),
+///            );
+///          },
+///        ),
+///      ),
+///    );
+///  }
+/// ```
+/// {@end-tool}
```

Comment:
I think we should not post BAD EXAMPLES in our docs. Somebody might accidentally copy this.

Reply:
This was added due to previous feedback, if I could get a definitive resolution on this that would be great

Reply:
We should only show good practices in our docs and remove the bad example.

Reply:
Thank you